Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.inl

```diff
@@ -183,6 +250,26 @@ namespace AZ::Dom
         return value == nullptr ? AZStd::forward<Deduced>(defaultValue) : *value;
     }
 
+    template<class T>
+    T& DomPrefixTree<T>::operator[](const Path& path)
+    {
+        Node* node = GetNodeForPath(path);
+        if (AZStd::is_default_constructible_v<T>)
```

Comment:
This should be `if constexpr` to force the compile time check.

Reply:
Ah yeah should get or create, sorted.